eth_account/_utils/typed_transactions.py

```diff
@@ -108,8 +125,11 @@ def from_dict(cls, dictionary: Dict[str, Any]):
             raise ValueError("missing or incorrect transaction type")
         # Switch on the transaction type to choose the correct constructor.
         transaction_type = pipe(dictionary['type'], hexstr_if_str(to_int))
+        transaction: Any
         if transaction_type == AccessListTransaction.transaction_type:
             transaction = AccessListTransaction
+        elif transaction_type == DynamicFeeTransaction.transaction_type:
```

Comment:
@malonaz I noticed another small update to the documentation. We should add the new `DynamicFeeTransaction` to the supported transaction types list in the docstring for the `TypedTransaction` class.